tools/testinfra/runner/src/main.rs

```diff
@@ -226,6 +295,129 @@ fn report<P: AsRef<Path>>(tests: Vec<TestResult>, path: P) -> Result<()> {
     return Ok(());
 }
 
-fn xml_escape_text(unescaped: String) -> String {
+fn xml_escape_text(unescaped: &String) -> String {
```

Comment:
`&str` would be more conventional, and `String/&String` can be dereferenced as a `&str`